pkg/management/barman/restorer/restorer.go

```diff
@@ -219,7 +220,12 @@ func (restorer *WALRestorer) RestoreList(
 
 // Restore restores a WAL file from the object store
 func (restorer *WALRestorer) Restore(walName, destinationPath string, baseOptions []string) error {
-	options := make([]string, len(baseOptions), len(baseOptions)+2)
+	optionsLength := len(baseOptions)
+	optionsLengthMax := optionsLength + 2
+	if optionsLength >= math.MaxInt || optionsLengthMax >= math.MaxInt {
```

Comment:
another redundant condition. `optionsLengthMax >= math.MaxInt` is enough ... unless codeql can't figure that out...